src/models/Event.js

```diff
@@ -39,27 +40,38 @@ const eventSchema = new Schema({
                     validator: (value) => validTime.test(value),
                     message: 'Invalid time format! Example: hh:mm (24h)'
                 }
-                
             }
-        }, { // Additional date and time (if needed)
-            date: { type: Date },
-            startTime: { type: String },
-            endTime: { type: String }
         }]
     },
-    imageUrl: { type: String, required: true },
+    imageUrl: {
+        type: String,
+        required: true,
+        validate: {
+            validator: (value) => validString.test(value),
+            message: "Invalid URL, must start with HTTP:///HTTPS://",
+        },
+    },
     contacts: {
         // TODO: Check later for unique COORDS..!
-        coordinates: { lat: { type: String }, long: { type: String } },
+        coordinates: { lat: { type: String, required: true }, long: { type: String, required: true } },
         city: { type: String, required: true },
         address: { type: String, required: true },
-        phone: { type: String, required: true }, 
+        phone: { type: String, required: true },
         email: { type: String, required: true }
     },
-    category: { type: String, required: true }, // Event category
-    likedCount: { type: Number, default: 0 }, // Count of users who liked the event
-    creator: { type: ObjectId, ref: 'User', required: true }, // User who created the event (you can specify user properties here)
-    winners: [{ name: { type: String } }, { name: { type: String } }, { name: { type: String } }], // Array of event winners (you can specify winner properties here)
+    category: {
+        type: String,
+        required: true,
+        minlength: [2, "Category must be minimum 2 characters long!"],
+        maxlength: [30, "Category must be maximum 30 characters long!"],
+    },
+    likes: [{ type: ObjectId, ref: 'User' } || { type: ObjectId, ref: 'Organization' }],
+    creator: { type: ObjectId, ref: 'User', required: true } || { type: ObjectId, ref: 'Organization', required: true },
```

Comment:
User cannot be a creator, only admin or organisation

Reply:
I will check later if user have admin property in controller function, also I dont have the updated version of user.

Reply:
We have not talk about admin model we talk about user Model with admin property............

Reply:
Which updated version of user? Make TODO’s so we don’t forget it later if you are going to check it later.